### PR TITLE
fix: [CDS-51757]: added RAW option to nexus3 artifact

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by sahithikolichala** [comment]
retrigger coverage

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger titlecheck

**Comment by sahithikolichala** [comment]
retrigger jest

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/NexusArtifact.tsx [hunk 1]
```diff
@@ -215,19 +215,22 @@ export function Nexus3Artifact({
   }, [repositoryDetails?.data])
 
   const getRepository = (): { label: string; value: string }[] => {
+    /* istanbul ignore next */
     if (fetchingRepository) {
       return [{ label: 'Loading Repository...', value: 'Loading Repository...' }]
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/NexusArtifact.tsx:220]: This text should come form yaml. Please update this.

File: src/modules/70-pipeline/utils/stageHelpers.ts [hunk 2]
```diff
@@ -91,6 +93,13 @@ export const nexus2RepositoryFormatTypes = [
   // { label: 'Raw', value: RepositoryFormatTypes.Raw }
 ]
 
+export const nexus3RepositoryFormatTypes = [
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/utils/stageHelpers.ts:96]: We can add common format and then add particular format. Common out nexus2 & nexus3 format and then add specific one/

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger coverage

**Comment by sahithikolichala** [comment]
retrigger jest

**Comment by sahithikolichala** [comment]
retrigger cypress-pipeline

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/__tests__/NexusArtifact.test.tsx [hunk 3]
```diff
@@ -385,4 +387,200 @@ describe('Nexus Artifact tests', () => {
...
+      </TestWrapper>
+    )
+
+    const repoFormatDropdown = container.querySelector('input[name="repositoryFormat"]') as HTMLInputElement
+    const repoFormatCaret = container
+      .querySelector(`input[name="repositoryFormat"] + [class*="bp3-input-action"]`)
+      ?.querySelector('[data-icon="chevron-down"]')
+    await waitFor(() => {
+      fireEvent.click(repoFormatCaret!)
+    })
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/__tests__/NexusArtifact.test.tsx:417]: `waitFor` is more suitable for waiting until assertions pass. In this case, you can wait for the caret to be in the document.

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/__tests__/NexusArtifact.test.tsx [hunk 4]
```diff
@@ -385,4 +387,200 @@ describe('Nexus Artifact tests', () => {
...
+    const repoFormatSelect = await findByText(container, 'Raw')
+    act(() => {
+      fireEvent.click(repoFormatSelect)
+    })
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/NexusArtifact/__tests__/NexusArtifact.test.tsx:421]: `fireEvent`/`userEvent` are already wrapped by `act`

**Review by rameshx — COMMENTED** [review]
Tests can be updated to follow best practices.

**Review by satyam-harness — APPROVED** [review]

**Pull request merged**